wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/StartEndCommand.java

```diff
@@ -27,19 +27,14 @@ public class StartEndCommand extends CommandBase {
    * @param requirements the subsystems required by this command
    */
   public StartEndCommand(Runnable onInit, Runnable onEnd, Subsystem... requirements) {
-    m_onInit = requireNonNullParam(onInit, "onInit", "StartEndCommand");
-    m_onEnd = requireNonNullParam(onEnd, "onEnd", "StartEndCommand");
-
-    addRequirements(requirements);
-  }
-
-  @Override
-  public void initialize() {
-    m_onInit.run();
-  }
-
-  @Override
-  public void end(boolean interrupted) {
-    m_onEnd.run();
+    super(
+        onInit,
+        () -> {},
+        // we need to do some magic here to null-check `onEnd` before it's captured
+        Optional.of(requireNonNullParam(onEnd, "onEnd", "StartEndCommand"))
+            .<Consumer<Boolean>>map(it -> (Boolean dropped) -> it.run())
+            .orElseThrow(),
```

Comment:
I think this be more easily understood as a private static function that just runs requireNonNull and then creates/returns the Runnable.